docs/parser.md

```diff
@@ -3,4 +3,91 @@ title: "Parser"
 weight: 90
 ---
 
->This document is under construction.
\ No newline at end of file
+## Parser
+
+AsyncAPI Parser is a package used to parse and validate the [AsyncAPI documents](asyncapi-document.md) in your Node.js or browser application. These documents can be either in YAML or JSON format.
+
+Parser validates these documents using dedicated schema-supported plugins. These supported schemas are:
+
+- AsyncAPI schema (no plugin needed)
+- OpenAPI schema
+- JSON schema
+- Avro schema
+- RAML data-type schema
+
+Parser allows the template developer to also easily access schemas provided in all of the above supported formats. This is because the JavaScript parser converts all of them into JSON schema.
+
+If the document is valid, Parser returns `AsyncAPIDocument` instance with a set of helper functions that enable easier access to contents of AsyncAPI document. The parser provides dereferenced output. During the dereference process the AsyncAPI parser substitutes a reference with a full definition. The dereferenced output is always in the JSON format. If a document is invalid, the parser provides a message listing all errors. The original asyncAPI document is part of the [Template Context](template-context.md) as generator passes the original asyncAPI document to the template context as well.	
+
+For example, the following asyncapi document has two channels–`channelOne` and `channelTwo`. Each channel has one operation and a single message:
+
+```yaml
+asyncapi: '2.5.0'
+info:
+  title: Demo API
+  version: '1.0.0'
+channels:
+  name/channelOne:
+    publish:
+      summary: This is the first sample channel
+      operationId: onMessage
+      message:
+        name: FirstMessage
+        payload:
+            id:
+              type: integer
+              minimum: 0
+              description: Id of the channel
+            sentAt:
+              type: string
+              format: date-time
+              description: Date and time when the message was sent.
+  name/channelTwo:
+    publish:
+      summary: This is the second sample channel
+      operationId: messageRead
+      message:
+        name: SecondMessage
+        payload:
+            id:
+              type: integer
+              minimum: 0
+              description: Id of the channel
+            sentAt:
+              type: string
+              format: date-time
+              description: Date and time when the message was sent.
+```
+We can use helper functions provided by parser to operate on the above JSON file. For example, we can use the helper method `asyncAPIDocument.channelNames()` which returns an array of all channel names currently present in the asyncAPI document. Another example where you can use a helper function is to list out messages present in your JSON file. Instead of fetching a single message one at a time, you can use the function `asyncAPIDocument.allMessages()` that returns the map of all messages present in your asyncAPI document.
```

Comment:
```suggestion
We can use helper functions provided by the Parser to operate on the above JSON file. For example, we can use the helper method `asyncAPIDocument.channelNames()`, which returns an array of all channel names currently present in the AsyncAPI document. Another example where you can use a helper function is to list out messages present in your JSON file. Instead of fetching a single message one at a time, you can use the function `asyncAPIDocument.allMessages()` that returns the map of all messages in your AsyncAPI document.
```